integration-tests/mongodb-panache/src/main/java/io/quarkus/it/mongodb/panache/book/BookEntityResource.java

```diff
@@ -0,0 +1,98 @@
+package io.quarkus.it.mongodb.panache.book;
+
+import java.net.URI;
+import java.time.LocalDate;
+import java.util.List;
+
+import javax.annotation.PostConstruct;
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.bson.types.ObjectId;
+import org.jboss.logging.Logger;
+
+import io.quarkus.panache.common.Parameters;
+import io.quarkus.panache.common.Sort;
+
+@Path("/books/entity")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class BookEntityResource {
+    private static final Logger LOGGER = Logger.getLogger(BookEntityResource.class);
+
+    @PostConstruct
+    void init() {
+        String databaseName = BookEntity.getMongoDatabase(BookEntity.class).getName();
```

Comment:
You looked at an outdated version.
It is now BookEntity.mongoDatabase() and there also exist BookEntity.mongoCollection().

Both are needed to be able to access low level API of MongoDB for eg to create indices, lauch aggregation, make updates ... And as the name of the collection and of the database can be customized in the `MongoEntity` annotation a user cannot directly rely on the mongoClient.getDatabase() method because it will then duplicate the name on the database ...

By the way, it makes me learn ASM so I really want to keep them ;)

Reply:
Yes.